### PR TITLE
add log/docker seeker

### PR DESCRIPTION
**Pull request by mkcp**
Added a seeker that checks if docker if available on the machine, and grabs logs from the given image name (currently called container, which docker has some fuzzy matching for). We also check if logs are empty in case nothing shows up in the time range.

While this is a good building block, there's going to be cases where there are many containers on a machine that are relevant and we need to grab logs for. One path forward is that we grab and write log files from _all_ containers that contain the product name.

### CODE REVIEW

File: product/consul.go [hunk 1]
```diff
@@ -43,6 +44,8 @@ func ConsulSeekers(tmpDir string, since, until time.Time) ([]*s.Seeker, error) {
 		s.NewHTTPer(api, "/v1/namespace"),
 		s.NewHTTPer(api, "/v1/status/leader"),
 		s.NewHTTPer(api, "/v1/status/peers"),
+
+		logs.NewDocker("consul", tmpDir, since, until),
```
Comment (nwchandler) [inline product/consul.go:48]: Will this assume that the running container is named "consul," or will Docker match based on image names?

Comment (mkcp) [reply product/consul.go:48]: Docker CLI calls the param `CONTAINER` but will happily match the image name too.

File: seeker/log/docker.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+// Docker allows logs to be retrieved for a docker container
+type Docker struct {
+	// Container is the name of the docker container to get logs from
+	Container string
+	// DestDir is the directory we will write the logs to
+	DestDir string
+	// Since marks the beginning of the time range to include logs
+	Since time.Time
+	// Until is not implemented yet, and is always "now"
```
[the remark this answers is not in the document]
Comment (mkcp) [reply seeker/log/docker.go]: Deleted it on current revision

File: product/consul.go (hunk 1 above)
Comment (nwchandler) [reply product/consul.go]: Nice! I did not know that.

**Review by nwchandler — APPROVED** [review]

**Pull request merged**